core/src/Cuda/Kokkos_Cuda_Error.hpp

```diff
@@ -63,17 +63,45 @@ void cuda_device_synchronize(const std::string& name);
 void cuda_stream_synchronize(const cudaStream_t stream,
                              const std::string& name);
 
-void cuda_internal_error_throw(cudaError e, const char* name,
-                               const char* file = nullptr, const int line = 0);
+[[noreturn]] void cuda_internal_error_throw(cudaError e, const char* name,
+                                            const char* file = nullptr,
+                                            const int line   = 0);
+
+[[noreturn]] void cuda_internal_error_abort(cudaError e, const char* name,
+                                            const char* file = nullptr,
+                                            const int line   = 0);
+
+#ifdef KOKKOS_ENABLE_CXX17
+#define KOKKOS_IMPL_FALLTHROUGH [[fallthrough]];
+#else
+#define KOKKOS_IMPL_FALLTHROUGH
+#endif
```

Comment:
I prefer this to be defined in `Kokkos_Macros.hpp` (if we need it).

Reply:
I removed it.